Sources/core/objects/Revolver.cpp

```diff
@@ -0,0 +1,65 @@
+#include "Revolver.h"
+
+Revolver::Revolver(size_t id) : Item(id)
+{
+    SetSprite("icons/guns.dmi");
+    SetState("revolver");
+
+    name = "Revolver",
+    
+    ammunition_ = 6;
+}
+
+bool Revolver::Handle_Ammo()
+{
+    if (ammunition_ <= 0)
+    {
+        ammunition_ = 0;
+        return false;
+    }
+    --ammunition_;
+    return true;
+}
+
+void Revolver::Shoot(VDir target)
+{	  
+    if(Handle_Ammo())
+    {
```

Comment:
Probably here is the issue. The owner of revolver in most cases is `Human`, not the `CubeTile`, so it is needed to make `Create<Projectile>(Bullet::T_ITEM_S(),GetOwner()->GetOwner());` (but it also better to check if the second owner is valid to be sure). Also, try to provide info about crashes - as far as I understand in this case some info must appear in stdout (something like "AddItem failed").